go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/ssvlabs/ssv
 
-go 1.22
+go 1.22.0
+
+toolchain go1.23.1
```

Comment:
Is there any reason why we need to have 1.23.1 patch version here and not 1.23.3 or 1.23.0?

Reply:
Not really, I don't think there is a specific reason.

It looks like it's just `go mod` behavior to add `toolchain` (if there is none, but `go 1.22` is specified and not `go 1.22.0`) with version that matches your local golang version (which is 1.23.1 for me),

this kind of behavior is a bit weird to me too, turns out I can just do `go 1.22.0` and omit `toolchain go1.23.1` completely (`go mod tidy` doesn't add it anymore when patch version is specified) - so I've done that to simplify things - https://github.com/ssvlabs/ssv/pull/1840/commits/de9e2f2fb52df6bc185ae207181182c0b6d55126